src/commands/admin/prune.ts

```diff
@@ -0,0 +1,111 @@
+import { Command } from '@lib/types/Command';
+import { ROLES } from '@root/config';
+import { botMasterPerms } from '@lib/permissions';
+import { GuildMember, Message, MessageActionRow, MessageButton, MessageEmbed } from 'discord.js';
+
+const PRUNE_TIMEOUT = 30;
+
+export default class extends Command {
+
+	description = `Prunes all members who don't have the <@&${ROLES.VERIFIED}> role`;
+	runInDM = false;
+
+	async permissions(msg: Message): Promise<boolean> {
+		return await botMasterPerms(msg);
+	}
+
+	async run(msg: Message): Promise<void> {
+		let timeout = PRUNE_TIMEOUT;
+		await msg.guild.members.fetch();
+		const toKick = msg.guild.members.cache.filter(member => !member.user.bot && !member.roles.cache.has(ROLES.VERIFIED));
+		if (toKick.size === 0) {
+			msg.channel.send('No prunable members.');
+			return;
+		}
+
+		const confirmEmbed = new MessageEmbed()
+			.setTitle(`Server prune will kick ${toKick.size} members from the guild. Proceed?`)
+			.setColor('RED')
+			.setFooter(`This command will expire in ${PRUNE_TIMEOUT}s`);
+
+		const confirmBtns = [
+			new MessageButton({ label: 'Cancel', customId: 'cancel', style: 'SECONDARY' }),
+			new MessageButton({ label: 'Proceed', customId: 'proceed', style: 'DANGER' })
+		];
+
+		const confirmMsg = await msg.channel.send({
+			embeds: [confirmEmbed],
+			components: [new MessageActionRow({ components: confirmBtns })]
+		});
+
+		const collector = msg.channel.createMessageComponentCollector({
+			filter: i => i.message.id === confirmMsg.id, time: PRUNE_TIMEOUT * 1000
+		});
+
+		const countdown = setInterval(() => this.countdown(confirmMsg, --timeout, confirmBtns, confirmEmbed), 1000);
+
+		collector.on('collect', async interaction => {
+			if (!interaction.isButton()) return;
+			if (interaction.user.id !== msg.author.id) {
+				await interaction.reply({
+					content: 'You cannot respond to a command you did not execute',
+					ephemeral: true
+				});
+				return;
+			}
+			interaction.deferReply({ ephemeral: true });
+			clearInterval(countdown);
+
+			confirmBtns.forEach(btn => btn.setDisabled(true));
+
+
+			if (interaction.customId === 'cancel') {
+				confirmEmbed.setColor('BLUE')
+					.setTitle(`Prune cancelled. ${msg.member.displayName} got cold feet!`);
+				confirmMsg.edit({ embeds: [confirmEmbed], components: [new MessageActionRow({ components: confirmBtns })] });
+			} else {
+				confirmEmbed.setTitle(`<a:loading:928003042954059888> Pruning ${toKick.size} members...`);
```

Comment:
What is the string of numbers in this embed? I know ':loading:' is the emoji you had Ren upload, but I'm unsure on what the numbers are for.

Reply:
That's the ID of the emote. Whenever you send an emote, you're actually sending `<:name:id>`. The `a` at the front is because it's an animated emote.